src/main/java/insat/company/platform/web/rest/IntegrationResource.java

```diff
@@ -38,30 +38,33 @@
 
     private final IntegrationService integrationService;
 
-    public IntegrationResource(IntegrationService integrationService) {
+    private final UserRepository userRepository;
+
+    public IntegrationResource(IntegrationService integrationService, UserRepository userRepository) {
         this.integrationService = integrationService;
+        this.userRepository = userRepository;
     }
 
     /**
      * POST  /integrations : Create a new integration.
      *
-     * @param integration the integration to create
+     * @param integrationDTO the integration to create
      * @return the ResponseEntity with status 201 (Created) and with body the new integration, or with status 400 (Bad Request) if the integration has already an ID
      * @throws URISyntaxException if the Location URI syntax is incorrect
      */
     @PostMapping("/integrations")
     @Timed
-    public ResponseEntity<Integration> createIntegration(@RequestBody Integration integration) throws URISyntaxException {
-        log.debug("REST request to save Integration : {}", integration);
-        if (integration.getId() != null) {
-            throw new BadRequestAlertException("A new integration cannot already have an ID", ENTITY_NAME, "idexists");
-        }
-        Integration result = integrationService.save(integration);
+    public ResponseEntity<Integration> createIntegration(@RequestBody IntegrationDTO integrationDTO) throws URISyntaxException {
+        log.debug("REST request to save Integration : {}", integrationDTO);
+        User user = userRepository.findOneWithAuthoritiesById(integrationDTO.getUserId()).get();
```

Comment:
you need to change this in case the user is not found, do Optional.ofNullable(integrationDTO.getUserId()).map(user->{
your logic.. 
}).orElseGet(()-> new ResponseEntity.badRequest("user not found"));